src/main/java/com/questhelper/helpers/quests/theheartofdarkness/TheHeartOfDarkness.java

```diff
@@ -0,0 +1,1028 @@
+                {
+                    tmpChars.add(matcher.group(1));
+                }
+            }
+        }
+
+        if (tmpChars.size() == 4)
+        {
+            inputArrows.setSolution(
+                    directionToInt.get(tmpChars.get(0)),
+                    directionToInt.get(tmpChars.get(1)),
+                    directionToInt.get(tmpChars.get(2)),
+                    directionToInt.get(tmpChars.get(3))
+            );
+
+            knowPoemSolution.setShouldPass(true);
+        }
+    }
+
+    @Override
+    public List<ItemRequirement> getItemRequirements()
+    {
+        return List.of(coins.quantity(30), combatGear, food, prayerPotions.quantity(2));
+    }
+
+    @Override
+    public List<ItemRequirement> getItemRecommended()
+    {
+        return List.of(quetzalFeed.quantity(10), limestoneBrick.quantity(3), softClay.quantity(4), staminaPotions);
+    }
+
+    @Override
+    public List<Requirement> getGeneralRequirements()
+    {
+        return List.of(
+                new SkillRequirement(Skill.MINING, 55),
+                new SkillRequirement(Skill.THIEVING, 48),
+                new SkillRequirement(Skill.SLAYER, 48),
+                new SkillRequirement(Skill.AGILITY, 46),
+                new QuestRequirement(QuestHelperQuest.TWILIGHTS_PROMISE, QuestState.FINISHED)
+        );
+    }
+
+    @Override
+    public List<String> getCombatRequirements()
+    {
+        return List.of(
+                "Several Emissary Brawlers (levels 74 and 86)",
+                "Several Emissary Conjurers (level 71)",
+                "Prince Itzla Arkan (level 167)",
+                "Amoxliatl (level 263)"
+        );
+    }
+
+    @Override
+    public QuestPointReward getQuestPointReward()
+    {
+        return new QuestPointReward(2);
+    }
+
+    @Override
+    public List<ExperienceReward> getExperienceRewards()
+    {
+        return List.of(
+                new ExperienceReward(Skill.MINING, 8000),
+                new ExperienceReward(Skill.THIEVING, 8000),
+                new ExperienceReward(Skill.SLAYER, 8000),
+                new ExperienceReward(Skill.AGILITY, 8000)
+        );
+    }
+
+    @Override
+    public List<UnlockReward> getUnlockRewards()
+    {
+        return Collections.singletonList(new UnlockReward("Access to Tapoyauik"));
+    }
+
+    @Override
+    public List<PanelDetails> getPanels()
+    {
+        List<PanelDetails> allSteps = new ArrayList<>();
+
+        allSteps.add(new PanelDetails("Starting off", List.of(talkToItzlaAtTeomat, travelToGorge, talkToBartender, restOnBed, talkToPrinceAfterRest,
+                talkToShopkeeper, talkToPrinceInPubAgain)));
+        allSteps.add(new PanelDetails("The Tower", List.of(talkToPrinceAtTower, buildSalvagerOverlookLandingSite, talkToPrinceAtTowerAfterLanding,
+                talkToNova, talkToSergius, talkToFelius, talkToCaritta, talkToPrinceAfterRecruits, talkToJanus, climbUpToFirstTrial), List.of(combatGear, food),
```

Comment:
talkToNova, talkToSergius, talkToFelius, talkToCaritta are listed in the reverse order here compared to the conditionalstep they're in